arch/arm/src/Makefile

```diff
@@ -157,7 +157,7 @@ board$(DELIM)libboard$(LIBEXT):
 	$(Q) $(MAKE) -C board libboard$(LIBEXT) EXTRAFLAGS="$(EXTRAFLAGS)"
 
 define LINK_ALLSYMS
-	$(Q) $(TOPDIR)/tools/mkallsyms.sh $(NUTTX) $(CROSSDEV) > allsyms.tmp
```

Comment:
$(CROSSDEV) is mandatory, otherwise nm will print some unnecessary information

Reply:
pyelftools has no options for cross-compiling toolchains，it is for parsing of generic elf files

Reply:
You need to compare the symbol output  of command nm, for example using arm-none-eabi-nm and  host "nm" are completely different

Reply:
There is a little difference between them, but I don't think it will make much difference

[nm.md](https://github.com/apache/incubator-nuttx/files/9665404/nm.md)
[pyelftools.md](https://github.com/apache/incubator-nuttx/files/9665405/pyelftools.md)

Reply:
Using nm you need to filter the text segment,
Seems that elftools can filter out the garbage symbols in the results you shared